token.cgi

```diff
@@ -509,6 +509,12 @@ sub mfa_event_from_token {
 
   # verify
   my $event = $user->mfa_provider->verify_token($token);
+
+  # If we got this far and MFA is Duo, we should be verified
+  if ($user->mfa eq 'Duo' && !$event->{duo_verified}) {
+    ThrowUserError('duo_user_error', {reason => 'Invalid Duo Security MFA Code'});
+  }
+
```

Comment:
Let's make a function for this and re-use it in `userprefs.cgi`.

Reply:
I am not sure that creating a function for this will gain much as it would be a very small check and is only in two places currently. Would the utility function throw the error or would it just return true or false. Would it do the $mfa eq 'Duo' check or would we do that before calling it. Just not sure it is worth the code line savings yet.

Reply:
The whole `if` block should be in one function like `assert_duo_verified($user->mfa, $event)`, since we're checking the exact same condition and throwing the exact same error in both places. It's mostly a maintainability suggestion, so if you're fine with maintaining the same logic in multiple places, it's no problem.